modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -43,134 +45,81 @@ resource "time_sleep" "wait_30_seconds" {
 }
 
 module "aws_provider_irsa" {
-  count                             = var.aws_provider.enable == true ? 1 : 0
+  count                             = local.aws_provider.enable == true ? 1 : 0
   source                            = "../../../modules/irsa"
   create_kubernetes_namespace       = false
   create_kubernetes_service_account = false
   kubernetes_namespace              = local.namespace
-  kubernetes_service_account        = "${local.aws_provider_sa}-*"
-  irsa_iam_policies                 = concat([aws_iam_policy.aws_provider[0].arn], var.aws_provider.additional_irsa_policies)
+  kubernetes_service_account        = "${local.aws_provider.name}-*"
+  irsa_iam_policies                 = local.aws_provider.additional_irsa_policies
   irsa_iam_role_path                = var.addon_context.irsa_iam_role_path
   irsa_iam_permissions_boundary     = var.addon_context.irsa_iam_permissions_boundary
   eks_cluster_id                    = var.addon_context.eks_cluster_id
   eks_oidc_provider_arn             = var.addon_context.eks_oidc_provider_arn
   depends_on                        = [kubectl_manifest.aws_provider]
 }
 
-resource "aws_iam_policy" "aws_provider" {
-  count       = var.aws_provider.enable == true ? 1 : 0
-  description = "Crossplane AWS Provider IAM policy"
-  name        = "${var.addon_context.eks_cluster_id}-${local.aws_provider_sa}-irsa"
-  policy      = data.aws_iam_policy_document.s3_policy.json
-  tags        = var.addon_context.tags
-}
-
 resource "kubectl_manifest" "aws_provider_config" {
-  count     = var.aws_provider.enable == true ? 1 : 0
-  yaml_body = templatefile("${path.module}/aws-provider/aws-provider-config.yaml", {})
+  count = local.aws_provider.enable == true ? 1 : 0
+  yaml_body = templatefile("${path.module}/aws-provider/aws-provider-config.yaml", {
+    aws-provider-config          = local.aws_provider.provider_config
+  })
 
   depends_on = [kubectl_manifest.aws_provider, time_sleep.wait_30_seconds]
 }
 
 #--------------------------------------
-# Terrajet AWS Provider
+# Kubernetes Provider
 #--------------------------------------
-resource "kubectl_manifest" "jet_aws_controller_config" {
```

Comment:
I'm not familiar with Crossplane but - are these resources that are being removed no longer used/required? What affect does this have on users who have already deployed Crossplane with the blueprints addon module - is this a breaking change for them?

Reply:
the jet provided is deprecated and it should not longer be use.
Its already removed from the example repo via this PR https://github.com/awslabs/crossplane-on-eks/pull/64

For folks that already deployed they should pin the version this repo.
How you proposed we move forward remove it like I have or put it back?

Reply:
@bryantbiggs I added the terrajet provider back